### PR TITLE
 0.2.1- Added overwrite function for upgraded plex after install. Fix…

### PR DESCRIPTION
**Pull request by liviynz**
0.2.1 - Added overwrite function for upgraded plex after install. Fixed chmod from 0777 to 0755 as that's a much safer approach to security. Fixed up some python format issues like double space before new function definition. Split out version to it's own function & fleshed out usage function. Added import of filecmp for file compare. Also some other minor warnings from analysis. - LiviyNZ - 06/01/2016

My testing so far has concluded the previous & new functions work as expected but please test yourself :)

I also have a hunch about the Fedora issue which is stopping the flavour slaves from working but will look into that later.

Cheers
Liviy

### CODE REVIEW

**Comment by peva3** [comment]
Are you having any issues with subtitles with your changes?

**Comment by liviynz** [comment]
Not that I've seen at all, I'm not sure why it would. That seems a very specific question to ask, I did notice your comment on the plex forums and that would be around the previous release that I'm looking to merge into. There's a few sticky issues that I've worked through since starting with this software and I think a lot of them have been due to the fact I'm using the latest plex rather than the version specified by the owner so that's to be expected since they've added quite a few new functions and changed things around. I've got specific issues open for them and have either put the fixes (if mentioned in the comment above) in this code or I'll address them separately. I have all functions that I'm aware of working fine on the 3 node system.

**Comment by wnielson** [comment]
Thanks for this.  A cursory glance through the code and things look pretty good, but I want to give it a more thorough look before pulling it.

**Comment by liviynz** [comment]
No worries, I'm quite confident as I was running it through an ide and then tested it in my test platform however Python is very new to me so I would have recommend a proper review anyway. :)

At some point this week I'll probably start a new branch for some of the other fixes/improvements I've manually done in my live system to get the current version of plex working across all 3 main transcoding types plus some other enhancements I have in mind.

**Comment by liviynz** [comment]
I did find one bug myself, erroneous if rather than a elif. It's fixed now and next version I'll probably swap out the argument if/else options for the built in python arguments function which is a bit cleaner & easier to manage.

**Comment by wnielson** [comment]
Things look good to me.  Can you make a few changes first though?
1. Remove the `# 0.2.1 - ...` comments
2. Add a contributors section to `README.md` and add yourself

Thanks!

File: prt.py [hunk 1]
```diff
@@ -1,4 +1,10 @@
 #!/usr/bin/env python
+# Version 0.2.0 - Original version - Weston Nielson <wnielson@github> - Date unknown
```
Comment (wnielson) [inline prt.py]: Remove these comments

File: prt.py [hunk 2]
```diff
@@ -11,6 +17,8 @@
 import subprocess
 import sys
 import time
+# 0.2.1 - Added filecmp for comparing 2 files
```
Comment (wnielson) [inline prt.py]: Remove

File: prt.py [hunk 3]
```diff
@@ -147,10 +163,42 @@ def install_transcoder():
     if rename_transcoder():
         try:
             shutil.copyfile(prt_remote, get_transcoder_path(ORIGINAL_TRANSCODER_NAME))
-            os.chmod(get_transcoder_path(ORIGINAL_TRANSCODER_NAME), 0777)
+            # 0.2.1 - Changed the chmod to 0775 rather than 0777 which is bad security practice
```
Comment (wnielson) [inline prt.py]: Remove

File: prt.py [hunk 4]
```diff
@@ -147,10 +163,42 @@ def install_transcoder():
...
+            os.chmod(get_transcoder_path(ORIGINAL_TRANSCODER_NAME), 0755)
         except Exception, e:
             print "Error installing new transcoder: %s" % str(e)
 
+
+# 0.2.1 - overwrite_transcoder_after_upgrade function - LiviyNZ
```
Comment (wnielson) [inline prt.py]: Remove

File: prt.py [hunk 5]
```diff
@@ -266,18 +315,39 @@ def transcode_remote():
     proc = subprocess.Popen(args)
     proc.wait()
 
+
+def version():
+    print "Plex Remote Transcoder version %s, Copyright (C) %s\n" % (__version__, __author__)
+
+
+# 0.2.1 - Fleshed out usage function
```
Comment (wnielson) [inline prt.py]: Remove

File: prt.py [hunk 6]
```diff
@@ -266,18 +315,39 @@ def transcode_remote():
...
+    "overwrite\t\tfixes PRT after PMS has had a version update breaking PRT\n" \
+    "add_host\t\tadds an extra host to the list of slaves PRT is to use\n" \
+    "remove_host\t\tremoves a host from the list of slaces PRT is to use\n"
+
 
 def main():
-    if len(sys.argv) < 2:
-        print "Plex Remote Transcoder version %s, Copyright (C) %s\n" % (__version__, __author__)
-        return usage()
+    # 0.2.1 - specific usage options
```
Comment (wnielson) [inline prt.py]: Remove

File: prt.py [hunk 7]
```diff
@@ -326,3 +396,21 @@ def main():
             print "Host removed"
         except Exception, e:
             print "Error removing host: %s" % str(e)
+
+    # 0.2.1 - added version option rathern than just for no options
```
Comment (wnielson) [inline prt.py]: Remove

File: prt.py [hunk 8]
```diff
@@ -326,3 +396,21 @@ def main():
...
+    elif any( [sys.argv[1] == "version", sys.argv[1] == "v", sys.argv[1] == "V"] ):
+        version()
+        sys.exit(0)
+
+    # 0.2.1 - added overwrite option (for after plex package update/upgrade)
```
Comment (wnielson) [inline prt.py]: Remove

File: prt.py [hunk 9]
```diff
@@ -326,3 +396,21 @@ def main():
...
+    elif sys.argv[1] == "overwrite":
+            overwrite_transcoder_after_upgrade()
+            print "Transcoder overwritten successfully"
+
+    # Todo: list_hosts option to show current hosts to aid add/remove_host options - Liviynz
+
+    # 0.2.1 - Anything not listed shows usage
```
Comment (wnielson) [inline prt.py]: Remove

**Comment by liviynz** [comment]
Will do shortly, I only added the comments to help highlight the changes and the reasons behind them

**Comment by wnielson** [comment]
@liviynz That is totally fine--you can keep the comments in there, just remove the version numbers from the comments.

**Comment by liviynz** [comment]
@wneilson Done (I think)...still getting used to this full Python IDE suite
with VCS built in :)

**Comment by wnielson** [comment]
Looks good, thanks!

**Comment by liviynz** [comment]
@wnielson Awesome! :) Moving forward is there a better way that we can communicate directly to collaborate on this or would you prefer to keep it on here. I just figured some less public way might be good for in depth discussions. If not that's fine.

**Pull request merged**